src/sketch.rs

```diff
@@ -250,26 +267,64 @@ pub fn sketch_files(
                                 .map(|it| Box::new(it) as Box<dyn RollHash>)
                                 .collect()
                         }
-                        _ => todo!(),
-                    };
-
-                    hash_its
-                        .iter_mut()
-                        .enumerate()
-                        .map(|(idx, hash_it)| {
-                            let sample_name = if concat_fasta {
-                                format!("{name}_{}", idx + 1)
+                        HashType::PDB => {
+                            if let Some(di) = &struct_strings {
+                                log::trace!("Length of string: {}", di.len());
+                                AaHashIterator::from_3di_string(di[idx].clone()) // TODO: clone is not ideal
+                                    .into_iter()
+                                    .map(|it| Box::new(it) as Box<dyn RollHash>)
+                                    .collect()
                             } else {
-                                name.to_string()
-                            };
-                            if hash_it.seq_len() == 0 {
-                                panic!("{sample_name} has no valid sequence");
+                                AaHashIterator::from_3di_file(fastx1, concat_fasta)
+                                    .into_iter()
+                                    .map(|it| Box::new(it) as Box<dyn RollHash>)
+                                    .collect()
                             }
-                            // Run the sketching
-                            // (&mut **? C++ called it wants its syntax back)
-                            Sketch::new(&mut **hash_it, &sample_name, k, sketch_size, rc, min_count)
-                        })
-                        .collect::<Vec<Sketch>>()
+                        }
+                    };
+
+
+                    // Temporal (?) restrictions to the size of proteins and chains
+                    if *seq_type == HashType::PDB {
```

Comment:
Could these checks go in `structures.rs` and be called before `from_3di_string` and `from_3di_file`?

Reply:
Yes... but you'll still need to modify something here (in sketch.rs) IF you still want to panic whenever you find a sequence with 0. The thing is that in this big iterate, you go over input_files: if you filter things in structures.rs, you might (you do) get a smaller struct_strings vector, and thus there is issue because you are expecting the same size as in input_files (as I was developing, I decided to not change the code so much and filter afterwards, despite being less efficient).

If we move the filtering to structures, we need to remove the panic somehow when treating PDB files (e.g. filtering those empty "sequences" in those cases), do you see it fit, or maybe you prefer other way? (We could also change more the iterator etc., but I'm not sure we want that...).

PS: I also discovered that the levels of aaHash were fixed to level1 with the PDB files: just removed them in case we want to explore there as we spoke in our last chat (always assuming the 3Di dict. keeps close the same symbols as those of the nucleotides)

Reply:
> PS: I also discovered that the levels of aaHash were fixed to level1 with the PDB files: just removed them in case we want to explore there as we spoke in our last chat (always assuming the 3Di dict. keeps close the same symbols as those of the nucleotides)

From memory I don't think the symbols correspond to similar motifs in the same way as AAs, but we should double-check in the foldseek paper

Reply:
Rather than filtering to a lower number of sequences, I think desired behaviour might be just to panic if anything fails and then the user can check, filter and rerun themselves (this is what worked best in PopPUNK)

Reply:
Ok, so:
- I checked Foldseek's paper/docs and they do not mention anything regarding similarity between symbols as with aminoacids, so I reverted the possibility to use different levels of aaHash.
- I removed the filters that I added, now I filter the pdb files before.
- Also removed the `concat_fasta` thing from previous comment.